### DISCUSSION TITLE
spi slave conflict

### DISCUSSION THREAD

**Discussion by hbrezner** [olikraus/u8g2#1725] [Q&A]
Hello:

I'm working with the u8x8 library, arduino, samd21 custom board. ssd1306 with this constructor
U8X8_SSD1306_64X48_ER_4W_SW_SPI oled(13, 11, 20, 7, 6);

both u8g2 and u8x8 work perfectly until I try to add in another slave device on the same spi bus, sercom1 in this case.

This might not be a u8x8 problem as the only way i could get my slave device working was using this method:
https://forum.arduino.cc/t/using-multiple-spi-buses-on-the-zero/635526/2

In my case the class instantiation is declared:
SPIClassSAMD mySPI (&sercom1, 12, 13, 11, SPI_PAD_0_SCK_1, SERCOM_RX_PAD_3);

i use these settings for mySPI
SPISettings settings(8000000, MSBFIRST, SPI_MODE0);

Oled output works fine until I issue this...
mySPI.begin();

thereafter, nothing appears on the screen.

**Comment by olikraus** [comment]
You probably need to use HW_SPI with u8x8. See also the u8g2 FAQ.

**Reply by hbrezner** [reply]
yes that's probably it. 

Need to rethink my spi pins on the next board revision. Currently I do not have hardware spi connected. The project involves one oled using your u8x8 lib, 1 sd card on spi, 1 spi slave and the i2s bus.

Originally I was going to have the oled and my slave device on the same spi bus and the sd card on its own but I'm wondering if I should have each on their own bus or rather have the oled on its own bus and have the sd card and slave sharing one bus.

any ideas?

**Comment by olikraus** [comment]
Individual pins for each device at least avoids any conflicts
